go-controller/pkg/ovn/controller/apbroute/repair.go

```diff
@@ -361,3 +377,67 @@ func (c *ExternalGatewayMasterController) buildOVNECMPCache() map[string][]*ovnR
 	}
 	return ovnRouteCache
 }
+
+// returns a hybrid cache of map[podIPs]nodeName
+func (c *ExternalGatewayMasterController) buildOVNHybridCache() map[string]string {
+	ovnHybridCache := make(map[string]string)
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.HybridOverlayReroutePriority
+	}
+
+	logicalRouterPolicies, err := c.nbClient.findLogicalRouterPoliciesWithPredicate(p)
+	if err != nil {
+		klog.Errorf("CleanHybridPRoutes: failed to list hybrid routes: %v", err)
+		return nil
+	}
+
+	var foundNextHops []string
+	for _, lrp := range logicalRouterPolicies {
+		foundNextHops = append(foundNextHops, lrp.Nexthops...)
+	}
+
+	r := func(item *nbdb.LogicalRouterPort) bool {
+		for _, nexthop := range foundNextHops {
+			for _, ip := range item.Networks {
+				// grab only IP prefix and not mask
+				p := strings.Split(ip, "/")
+				if p[0] == nexthop {
+					return true
+				}
+			}
+		}
+		return false
+	}
```

Comment:
I would be a fan of using `k8s.io/apimachinery/pkg/util/sets` for `foundNextHops` but up to you, we can potentially save cycles if there is chance for it to have dups

Reply:
ok i will change it